### PR TITLE
Fixes #266: added test to ensure arrives to the Neo4j Driver

### LINKED ISSUE #266: Verify LDAP Basic Authentication
See https://neo4j.com/docs/driver-manual/current/client-applications/

Authentication type basic can work with LDAP, but need to verify how the token is formed and how that is expressed in terms of login.

### PR DESCRIPTION
**Pull request by utnaf**
Fixes #266 

Tried to check that the correct token is injected into the `ConfigurationSettings` [class](https://github.com/neo4j/neo4j-java-driver/blob/4.3/driver/src/main/java/org/neo4j/driver/internal/DriverFactory.java#L106) but wasn't able to.

Here we ensure that the driver is created properly by the `GraphDatabase` class.

### CODE REVIEW

File: common/src/test/java/org.neo4j.spark.service/AuthenticationTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.times;
+
+@PrepareForTest({GraphDatabase.class})
+@RunWith(PowerMockRunner.class)
+public class AuthenticationTest {
+
+    @Test
+    public void testLdapConnectionToken() {
```
Comment (conker84) [inline common/src/test/java/org.neo4j.spark.service/AuthenticationTest.java]: @utnaf can we write this in scala?

Comment (utnaf) [reply common/src/test/java/org.neo4j.spark.service/AuthenticationTest.java]: Nope, can get it working in Scala, sadly there is an issue with power mock that doesn't make it work

Comment (utnaf) [reply common/src/test/java/org.neo4j.spark.service/AuthenticationTest.java]: Jk, it was the other test I tried to do that wasn't working in scala, this one is. Will move it to the scala package.

File: test-support/pom.xml [hunk 2]
```diff
@@ -49,6 +53,18 @@
             <version>2.12.0</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-module-junit4</artifactId>
+            <version>${powermock.version}</version>
```
Comment (conker84) [inline test-support/pom.xml:59]: add both as test deps

File: test-support/pom.xml [hunk 3]
```diff
@@ -49,6 +53,18 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-api-mockito2</artifactId>
+            <version>${powermock.version}</version>
```
Comment (conker84) [inline test-support/pom.xml:65]: test dep

**Review by conker84 — CHANGES_REQUESTED** [review]

**Comment by conker84** [comment]
LGTM thank you @utnaf

**Pull request merged**